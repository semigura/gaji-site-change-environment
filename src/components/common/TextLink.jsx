import classNames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
import { Link } from 'gatsby';
import Icon from './icons/Icon';

import css from './TextLink.module.scss';

const TextLink = props => {
  const {
    className,
    isCenter,
    isLeft,
    isRight,
    label,
    onClickLink,
    to,
    withIcon,
  } = props;
  const classes = classNames(
    css.TextLink,
    className,
    withIcon ? css.withIcon : '',
    isCenter ? css.isCenter : '',
    isLeft ? css.isLeft : '',
    isRight ? css.isRight : '',
  );
  const iconName = isLeft ? 'arrow-left' : 'arrow-right';

  return (
    <div className={classes}>
      {to ? (
        <Link className={css.TextLink__Link} to={to}>
          {label}
          {withIcon && (
            <Icon
              className={css.TextLink__Icon}
              name={iconName}
              aria-hidden="true"
            />
          )}
        </Link>
      ) : (
        <button
          className={css.TextLink__Link}
          type="button"
          onClick={onClickLink}
        >
          {label}
          {withIcon && (
            <Icon
              className={css.TextLink__Icon}
              name={iconName}
              aria-hidden="true"
            />
          )}
        </button>
      )}
    </div>
  );
};

TextLink.propTypes = {
  className: PropTypes.string,
  isCenter: PropTypes.bool,
  isLeft: PropTypes.bool,
  isRight: PropTypes.bool,
  label: PropTypes.string.isRequired,
  onClickLink: PropTypes.func,
  to: PropTypes.string,
  withIcon: PropTypes.bool,
};

TextLink.defaultProps = {
  className: undefined,
  isCenter: false,
  isLeft: false,
  isRight: false,
  onClickLink: undefined,
  to: undefined,
  withIcon: false,
};

export default TextLink;
