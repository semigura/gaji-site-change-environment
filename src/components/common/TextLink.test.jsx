import { shallow } from 'enzyme';
import React from 'react';

import TextLink from './TextLink';

describe('<TextLink />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<TextLink label="Label" />);
    expect(wrapper).toMatchSnapshot();
  });

  describe('isCenter', () => {
    it('Class name `isCenter` is applied', () => {
      const wrapper = shallow(<TextLink label="Label" isCenter />);
      expect(wrapper.find('.isCenter')).toHaveLength(1);
    });
  });

  describe('withIcon', () => {
    it('renders correctly', () => {
      const wrapper = shallow(<TextLink label="Label" withIcon />);
      expect(wrapper.find('.withIcon')).toHaveLength(1);
      expect(wrapper.find('.TextLink__Icon')).toHaveLength(1);
    });
  });

  describe('isRight', () => {
    it('Class name `isRight` is applied', () => {
      const wrapper = shallow(<TextLink label="Label" withIcon isRight />);
      expect(wrapper.find('.isRight')).toHaveLength(1);
    });
  });

  describe('isLeft', () => {
    it('Class name `isRight` is applied', () => {
      const wrapper = shallow(<TextLink label="Label" withIcon isLeft />);
      expect(wrapper.find('.isLeft')).toHaveLength(1);
    });
  });
});
