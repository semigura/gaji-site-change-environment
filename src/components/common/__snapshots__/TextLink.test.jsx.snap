// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextLink /> renders correctly 1`] = `
<div
  className="TextLink"
>
  <button
    className="TextLink__Link"
    type="button"
  >
    Label
  </button>
</div>
`;
